dali/operators/image/morphology/morphology.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/morphology/morphology.h"
+
+namespace dali {
+
+bool Morphology::SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) {
+  const auto &input = ws.Input<GPUBackend>(0);
+  auto sh = input.shape();
+  output_desc.resize(1);
+  output_desc[0] = {sh, input.type()};
+  return true;
+}
+
+void Morphology::RunImpl(Workspace &ws) {
+  const auto &input = ws.Input<GPUBackend>(0);
+  auto &output = ws.Output<GPUBackend>(0);
+  output.SetLayout(input.GetLayout());
+
+  kernels::DynamicScratchpad scratchpad({}, AccessOrder(ws.stream()));
+  auto mask = AcquireTensorArgument<int32_t, NVCV_DATA_TYPE_2S32>(ws, scratchpad, mask_arg_,
+                                                                  TensorShape<1>(2));
+  auto anchor = AcquireTensorArgument<int32_t, NVCV_DATA_TYPE_2S32>(ws, scratchpad, anchor_arg_,
+                                                                    TensorShape<1>(2));
+
+  if (workspace_.capacity() < input.num_samples()) {
```

Comment:
```suggestion
  if (!workspace_ || workspace_.capacity() < input.num_samples()) {
```
...and get rid of the (usually incorrect) initialization in the constructor.

Reply:
Actually, I couldn't make it work. It thrown the invalid handle error